tests/fused_attention_stablehlo_test.py

```diff
@@ -107,55 +154,50 @@ def sdpa_train_ref(query: Array,
             bias: Optional[Array] = None,
             mask: Optional[Array] = None,
             scale: float = 0.5,
-            is_causal_mask: bool = False,
+            mask_type: MaskType = MaskType.NO_MASK,
             dropout_rate: float = 0.1) -> Array:
   out_ref, sdpa_vjp_ref = jax.vjp(
-    partial(sdpa_ref, scale=scale, is_causal_mask=is_causal_mask, dropout_rate=dropout_rate),
+    partial(
+      sdpa_ref, scale=scale, mask_type=mask_type, dropout_rate=dropout_rate),
     query, key, value, bias, mask)
-  query_grad_ref, key_grad_ref, value_grad_ref, _, _ = sdpa_vjp_ref(grad)
+  query_grad_ref, key_grad_ref, value_grad_ref, bias_grad_ref, _ = sdpa_vjp_ref(grad)
+  if bias is not None and len(bias.shape) == 3:
+    return out_ref, (query_grad_ref, key_grad_ref, value_grad_ref, bias_grad_ref)
   return out_ref, (query_grad_ref, key_grad_ref, value_grad_ref)
 
 class DotProductAttentionTest(jtu.JaxTestCase):
   def setUp(self):
-    # TODO(Cjkkkk): Tests fail when the cuDNN constraints check fails, or
-    #  even when it passes: https://github.com/google/jax/issues/20438
-    self.skipTest('Failing with various errors')
-    return
-
+    super().setUp()
     if jax.device_count() < 4:
       self.skipTest("Requires more than 4 devices.")
     try:
       cudnn_version = check_cudnn_version()
+      check_compute_capability((80, 90))
```

Comment:
FYI we have `jtu.is_cuda_compute_capability_at_least` which does this.

Reply:
cudnn flash attn only support 8.0, 9.0 not 8.6. So at least is not enough.

Reply:
Sure, my point was that you can use a `jtu` API to do the same check.